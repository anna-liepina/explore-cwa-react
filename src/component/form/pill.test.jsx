import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Pill from './pill';

describe('<Pill/>', () => {
    const props = {
        label: 'props.label',
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        'data-id': 'optProps.data-id',
        className: 'optProps.className',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<Pill {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<Pill {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
