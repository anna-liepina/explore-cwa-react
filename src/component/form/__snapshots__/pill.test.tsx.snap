// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill/> render with default/required props 1`] = `
<DocumentFragment>
  <span
    class="pill"
    data-cy=""
  >
    props.label
  </span>
</DocumentFragment>
`;

exports[`<Pill/> render with optional/required props 1`] = `
<DocumentFragment>
  <span
    class="pill optProps.className"
    data-cy="optProps.data-cy"
  >
    props.label
    <button
      class="pill__control"
      data-cy="optProps.data-cy-remove"
      data-id="optProps.data-id"
    >
      Ã—
    </button>
  </span>
</DocumentFragment>
`;
